analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -0,0 +1,290 @@
+        val contents = Toml().read(lockfile)
+        val metadata = contents.getTable("metadata") ?: return emptyMap()
+        val metadataMap = metadata.toMap()
+        return metadataMap.asSequence().mapNotNull {
+            val key = it.key
+            val value = it.value as? String
+            if (key != null && value != null) {
+                Pair(key, value)
+            } else {
+                null
+            }
+        }.toMap()
+    }
+
+    private fun resolveDependenciesTree(
+        metadata: JsonNode,
+        packages: Map<String, Package>,
+        filter: (pkgId: String, depId: String) -> Boolean = { _, _ -> true }
+    ): SortedSet<PackageReference> {
+        val resolve = metadata["resolve"]
+        val nodes = resolve["nodes"]
+        val root = resolve["root"].textValueOrEmpty()
+
+        return resolveDependenciesOf(
+            root, nodes, packages, filter
+        ).dependencies
+    }
+
+    private fun resolveDependenciesOf(
+        id: String,
+        nodes: JsonNode,
+        packages: Map<String, Package>,
+        filter: (pkgId: String, depId: String) -> Boolean
+    ): PackageReference {
+        val node = nodes.find { it["id"].textValueOrEmpty() == id }!!
+        val depsReferences = node["dependencies"].asSequence()
+            .map { it.textValue()!! }
+            .filter { filter(id, it) }
+            .map { resolveDependenciesOf(it, nodes, packages, filter) }
+        val pkg = packages[id]!!
+        val linkage = if (isProjectDependency(id)) PackageLinkage.PROJECT_STATIC else PackageLinkage.STATIC
+        return pkg.toReference(linkage, dependencies = depsReferences.toSortedSet())
+    }
+
+    private fun isDevDependencyOf(id: String, depId: String, metadata: JsonNode): Boolean {
+        val packages = metadata["packages"]
+        val pkg = packages.asSequence().find { it["id"].textValueOrEmpty() == id }!!
+        val depPkg = packages.asSequence().find { it["id"].textValueOrEmpty() == depId }!!
+        val dep = pkg["dependencies"].find {
+            val name = it["name"].textValueOrEmpty()
+            name == depPkg["name"].textValueOrEmpty()
+        }!!
+        return dep["kind"].textValueOrEmpty() == "dev"
+    }
+
+    // Check if a package is a project
+    //
+    // We treat all path dependencies inside of the analyzer root as project dependencies
+    private fun isProjectDependency(id: String): Boolean {
+        val pathRegex = Regex("""^.*\(path\+file://(.*)\)$""")
+        val match = pathRegex.matchEntire(id)?.groups?.get(1)
+        if (match != null) {
+            val packageDir = File(match.value)
+            return packageDir.startsWith(analyzerRoot)
+        }
+        return false
+    }
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        log.info { "Resolving dependencies for: '$definitionFile'" }
+
+        // get project name; if none => we have a workspace definition => return null
```

Comment:
Nit: Comments should start with a capital letter and end with a dot, just like written sentences.